src/main/java/me/contaria/seedqueue/SeedQueueThread.java

```diff
@@ -32,12 +32,15 @@ public void run() {
             try {
                 // clear pinged state when starting a new check
                 this.pinged.set(false);
-
                 if (SeedQueue.shouldPauseGenerating()) {
                     this.pauseSeedQueueEntry();
                     continue;
                 }
-                if (!SeedQueue.shouldGenerate()) {
+                if (!SeedQueue.shouldGenerate() || SeedQueue.shouldResumeAfterQueueFull()) {
+                    if (!SeedQueue.shouldResumeGenerating() && !SeedQueue.noLockedRemaining()) {
+                        this.pauseSeedQueueEntry();
+                        continue;
+                    }
                     if (SeedQueue.shouldResumeGenerating() && this.unpauseSeedQueueEntry()) {
```

Comment:
Please reduce this to one call to `SeedQueue.shouldResumeGenerating()` to avoid any potential concurrency issues causing the calls to return different results. I'm not sure if it would cause any failure but no need to risk it

Reply:
Gotcha so I just add a Boolean var to store it....

Reply:
Wait this doesn't really matter because I am `continue`ing in both of those branches right? It would behave as an if-else basically if shouldResumeGenerating() is either false or true respectively. I will still refactor tho, because it makes more sense to do so...